### PR TITLE
feat: authentication

### PR DESCRIPTION
**Pull request by Nirajn2311**
This PR implements authentication and the required keys are securely stored for later use. The temporary login view is hidden behind a flag(similar to the forum) so that the PR can be merged and work can be split among contributors.

Ref: #11

### CODE REVIEW

File: mobile-app/lib/service/authentication_service.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          'https://www.freecodecamp.dev/learn/?messages=success%5B0%5D%3Dflash.signin-success') {
+        isLoggedIn = true;
+        Map<String, String> cookies = await browser.getCookies();
+        // Do not question the weird keys below
+        _csrf = cookies['"_csrf']!;
+        _csrfToken = cookies[' csrf_token']!;
+        _jwtAccessToken = cookies[' jwt_access_token']!;
+        log('_csrf: $_csrf');
+        log('_csrfToken: $_csrfToken');
+        log('_jwtAccessToken: $_jwtAccessToken');
```
Comment (Sembauke) [inline mobile-app/lib/service/authentication_service.dart:47]:
```suggestion
        showKeys(); 
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: mobile-app/lib/ui/widgets/drawer_widget/drawer_widget_view.dart [hunk 2]
```diff
@@ -86,6 +86,15 @@ class DrawerWidgetView extends StatelessWidget {
                       icon: Icons.favorite,
                     ),
                     buildDivider(),
+                    model.showForum
```
[the remark this answers is not in the document]
Comment (Nirajn2311) [reply mobile-app/lib/ui/widgets/drawer_widget/drawer_widget_view.dart:89]: I made it a separate page as then we could add extra buttons which would help in debugging when we come across any issue.

File: mobile-app/lib/service/authentication_service.dart (hunk 1 above)
Comment (Nirajn2311) [reply mobile-app/lib/service/authentication_service.dart:47]: Ahh yes good catch

**Comment by Sembauke** [comment]
The test failed but it doesn't really matter.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**